### ISSUE TITLE
TcNo-Acc-Switcher still doesn't kill procesess

### ISSUE THREAD

**Issue by supertrio17** [TcNobo/TcNo-Acc-Switcher#116]
![image](https://user-images.githubusercontent.com/57882120/145650136-06cce49e-14af-453f-a927-5dcfa51bade6.png)
Still the same. I updated with updater but had the same issue, so i uninstalled and downloaded latest version from github, and still doesn't work. Installed version is 2021-12-10_00

**Comment by TcNobo** [comment]
Mine seems ot work fine... If you close everything, and start the tray as admin, does it work as expected? 
I have an idea of another way to fix this issue though it will take some figuring out

**Comment by supertrio17** [comment]
I tried running it as admin, but still same thing.

**Comment by supertrio17** [comment]
And I wanted to ask, what is the reason behind duplicate .exe files in installation folder. Like "TcNo-Acc-Switcher-Tray.exe" and "TcNo-Acc-Switcher-Tray_main.exe"?

**Comment by TcNobo** [comment]
I'm working on a fix now.
TLDR: I updated from .NET 5 to .NET 6 as it has long term support etc.
- ~40k exisitng users can update automatically... but will all get the popup saying they need to download "ASP.NET" and another for "desktop runtime" etc etc... Why not install it automatically?
- That's what I did. A "wrapper" for each exe so they can run anything, it verifies they have .NET 6 installed, then launches the app as normal.
- This also fixes the "why am i getting this popup bro?" when instructions are as simple as I can make them...

This does of course introduce issues like this ^

I do want to figure out how the normal .exe works, because in .NET the .dll has the program in it. Run `dotnet run program.dll` and the program runs. The .exe just launches it. If I run the command `dotnet run program.dll` it starts a `dotnet.exe` process with the program running inside it, making it difficult to find and kill for updates, from the tray etc.
I ended up keeping the original .exe, and just replacing the normal ones people have shortcuts to with the wrapper.

This has fixed multiple issues that have long plagued me, such as "why popup!?!?!"... But have broken a few features like this ^
I should have already fixed this with my last commit though it needs some tweaking :)

**Comment by supertrio17** [comment]
Understood, thank you for explaining it to me :)

**Comment by supertrio17** [comment]
I also found out that when trying to kill processes with holding alt and starting TcNo-Acc-Switcher it doesn't work. It starts both "TcNo-Acc-Switcher.exe" and "TcNo-Acc-Switcher_main.exe" and then kills just them.

**Comment by supertrio17** [comment]
I found a message in Event Viewer from today while I was experiencing problems with TcNo, when it said that I don't have Net 6 runtime, tought it might help you.
![image](https://user-images.githubusercontent.com/57882120/145723875-45968f50-937c-416c-a040-6cf7b1c574ff.png)
And I get this warning every time I open Acc Switcher
![image](https://user-images.githubusercontent.com/57882120/145724021-3d91cc6c-a027-4824-99e0-9b8b9a789994.png)

**Comment by TcNobo** [comment]
The first I assume is once-off and related to the popup you get asking you to install. 
The second looks interesting.. I'll look into it ^ What Windows are you on? Win 10 Home? Pro? etc.

Finally, Download the latest release candidate from here (a beta version): https://github.com/TcNobo/TcNo-Acc-Switcher/releases/tag/2021-12-12_00
The tray should be the program's "parent", and close them on exit. Whereas before they were "detatched".
Let me know if this issue is fixed... It should be :)

**Comment by supertrio17** [comment]
I'm using Windows 10 Pro. Regarding this problem, I uninstalled the TcNo and installed the beta version. It seems that now it's not even trying to open the program when double clicking on tray icon. Issue is still there, when closing, all processes are still there. When trying to force close "TcNo-Acc-Switcher.exe" by force, it closes the tray as well, maybe instead of tray being the "parent" it's the main program. And also, there is no popup when trying to open TcNo from tray, it seems that it doesn't do anything.

**Comment by supertrio17** [comment]
One more minor fix you can look into is when installer finishes you get promted cmd that says press to continue, you press and program opens but cmd doesn't close. Nothing that should impact program performance.

**Comment by supertrio17** [comment]
I did a little bit more testing, and here is what I have. So I tried running the command "/C TASKKILL /F /T /IM {procName}" from cmd, everything worked as expected except that "/C" is not recognised as argument. I'm not sure if that has something to do with problem. 

Other than that, I tried closing without 'Exit minimizes to tray', and it worked as it should, so I don't see why isn't it working. Shouldn't program close the same even without that setting enabled?

**Comment by TcNobo** [comment]
I'll look even further into it :/ I think I know what's happening with the installer as well.
Just ignore the /C, and the command should kill whatever you put into `{procName}`, eg: `steam`

**Comment by supertrio17** [comment]
Okay, I'll write here if I gather some more info for you.

**Comment by supertrio17** [comment]
Um, it suddenly started working. Did you push an update or did I do something? If I did, I can try and find out what I did if somebody else has the same issue.

**Comment by TcNobo** [comment]
2021-12-12_00 was the last update I released, and it's a pre-release meaning you had to download it straight from Discord.
If you manually start the tray it should stay open, or if you change settings to disable it after launch.

I'll test it on my side to see if things don't work as expected :)

**Comment by supertrio17** [comment]
Any updates on this issue so far? For the time being I disabled Exit to tray and startup tray, and I created shortcut inside stratup folder itself. If anyone experiences similar problems they can do this as well.

**Comment by TcNobo** [comment]
There is now another method I implemented that should work a lot better. If it doesn't, there is an option in the Settings for each platform that allows you to use a different method instead -- Though this alternate method is the one we were having issues with here. If it's not released yet, it should be included with the next major update coming soon.

See the above link for more info.
